### PR TITLE
week6/assignment6: finding the 3 element sum closest to the target 

### PR DESCRIPTION
**Pull request by tusharwalzade216**
## 3 sum

Given an array S of n integers, find three integers in S such that the sum is closest to a given number, target. Return the sum of the three integers. Assume that there will only be one solution.

Example:

given array S = {-1 2 1 -4}, and target = 1.

The sum that is closest to the target is 2. (-1 + 2 + 1 = 2)

### CODE REVIEW

File: Week-6/Assignment-6/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ */
+const getClosestSum = (arr, target) => {
+    if (!validateInput(arr, target)) throw new Error('Invalid input provided');
+
+    const arrLen = arr.length;
+    let startIndex, endIndex = arrLen - 1;
+    let closestSum = Infinity;
+
+    // sorting the array first to have better time complexity
+    for (let i = 0; i < arrLen; i++) {
```
Comment (sairamin) [inline Week-6/Assignment-6/index.js]: Why can't you use inbuilt sorting mechanisms ?

File: Week-6/Assignment-6/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        for (let j = 0; j < arrLen; j++) {
+            if (arr[j] > arr[j + 1]) {
+                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
+            }
+        }
+    }
+
+    // go upto start of the last triplet, i.e. 3rd last value
+    for (let i = 0; i < arrLen - 2; i++) {
+        startIndex = i + 1;
```
Comment (sairamin) [inline Week-6/Assignment-6/index.js:35]: Don't you need to reinitialise endIndex for each loop ?

**Review by sairamin — CHANGES_REQUESTED** [review]
Need some clarity

File: Week-6/Assignment-6/index.js (hunk 1 above)
Comment (tusharwalzade216) [reply Week-6/Assignment-6/index.js]: Thanks @sairamin, using arr.sort now, please review

File: Week-6/Assignment-6/index.js (hunk 2 above)
Comment (tusharwalzade216) [reply Week-6/Assignment-6/index.js:35]: Thanks @sairamin, but as per my understanding it's not required.

**Justification** - We already use the sorted array and if calculated sum > target sum, implies that the solution lies in the left of the end and there is no point doing addition of greater values positioned at the end. We are reducing the data by conditionally incrementing startIndex and decrementing endIndex here. So, updated endIndex will be used in this case.

Please correct me if my understanding is wrong.

**Review by sairamin — APPROVED** [review]

**Pull request merged**